.jenkins

```diff
@@ -37,11 +37,20 @@ pipeline {
                             args '-v /tmp/ccache.kokkos:/tmp/ccache --env NVIDIA_VISIBLE_DEVICES=$NVIDIA_VISIBLE_DEVICES'
                         }
                     }
+                    environment {
+                        OMP_NUM_THREADS = 8
+                        // Nested OpenMP does not work for this configuration,
+                        // so disabling it
+                        OMP_MAX_ACTIVE_LEVELS = 1
+                        OMP_PLACES = 'threads'
+                        OMP_PROC_BIND = 'spread'
+                        NVCC_WRAPPER_DEFAULT_COMPILER = 'nvc++'
+                    }
                     steps {
                         sh '''rm -rf build && mkdir -p build && cd build && \
                               /opt/cmake/bin/cmake \
                                 -DCMAKE_BUILD_TYPE=Debug \
-                                -DCMAKE_CXX_COMPILER=nvc++ \
+                                -DCMAKE_CXX_COMPILER=$WORKSPACE/bin/nvcc_wrapper \
```

Comment:
Can you please clarify why this changed? I thought you wanted to test `nvc++` instead of `nvcc_wrapper` as `CUDA` compiler.

Reply:
With this PR, we would now be using nvc++ to compile CUDA code which is not quite fully ready.  So here I select the mvcc_wrapper with nvc++ as the host compiler to keep this build as it was before.

Reply:
In this case, I would expect `KOKKOS_CXX_COMPILER_ID` to be `NVIDIA` and `KOKKOS_CXX_HOST_COMPILER_ID` to be `NVHPC` but all the changes here are for `KOKKOS_CXX_COMPILER_ID=NVHPC`.
What am I missing?

Reply:
Correct.  There is currently no build that test compiling with NVC++ but that is ok.
All the "do not rely on `ifdef __CUDA_ARCH__`" changes are getting us closer to it.  The main blocker is we need a major update to the Desul atomics.  Christian and I are looking into it.

Reply:
I think in that case I would wait with this pull request until we can use `nvc++` as the device compiler in the CI. BUt I won't block if others want this to be merged already.